File: feature-favorite-venues/src/main/java/home/oleg/feature_favorite_venues/VenueViewItem.java
package home.oleg.feature_favorite_venues;

import java.util.ArrayList;
import java.util.List;

import home.oleg.placesnearme.core_presentation.viewdata.VenueViewData;

public class VenueViewItem implements ItemViewType {

    public static int VIEW_TYPE = 1;

    private final VenueViewData venueViewData;

    public VenueViewItem(VenueViewData venueViewData) {
        this.venueViewData = venueViewData;
    }

    public VenueViewData getVenueViewData() {
        return venueViewData;
    }

    @Override
    public int getViewType() {
        return VIEW_TYPE;
    }

    public static List<VenueViewItem> map(List<VenueViewData> venueViewDataList) {
        List<VenueViewItem> venueViewItems = new ArrayList<>();

        for (VenueViewData venueViewData : venueViewDataList) {
            venueViewItems.add(new VenueViewItem(venueViewData));
        }

        return venueViewItems;
    }


}

File: core-presentation/src/main/java/home/oleg/placesnearme/corepresentation/recyclerview/VenueViewItem.kt
package home.oleg.placesnearme.corepresentation.recyclerview

import home.oleg.placesnearme.corepresentation.viewdata.PlaceViewData
import java.util.*

data class VenueViewItem(val venueViewData: PlaceViewData) : com.smedialink.common.recyclerview.ItemViewType {

    override fun getViewType(): Int {
        return VIEW_TYPE
    }

    companion object {

        var VIEW_TYPE = 1

        fun map(venueViewDataList: List<PlaceViewData>): List<VenueViewItem> {
            val venueViewItems = ArrayList<VenueViewItem>()

            for (venueViewData in venueViewDataList) {
                venueViewItems.add(VenueViewItem(venueViewData))
            }

            return venueViewItems
        }
    }
}
